src/services/resource.js

```diff
@@ -1178,4 +1182,96 @@ module.exports = class resourceHelper {
 
 		return orgConfigList[resourceType]
 	}
+
+	/**
+	 * Get resources from consumption service
+	 * @name browseExistingList
+	 * @param {String} organization_id - Org Id of the user
+	 * @param {String} token - Token of the user
+	 * @param {Object} query - Query object passed by user
+	 * @param {String} searchText - Title to search
+	 * @param {Integer} pageNo -  Used to skip to different pages. Used for pagination . If value is not passed, by default it will be 1
+	 * @param {Integer} pageSize -  Used to limit the data. Used for pagination . If value is not passed, by default it will be 100
+	 * @returns {Object} - Response contain object of user details
+	 */
+	static async browseExistingList(organization_id, token, query, searchText = '', pageNo, pageSize) {
+		try {
+			let result = {
+				data: [],
+				count: 0,
+			}
+			const type = query[common.TYPE] ? query[common.TYPE] : ''
+			const search = searchText != '' ? searchText : ''
+			let resources = {}
+			if (process.env.CONSUMPTION_SERVICE != common.SELF) {
+				resources = await interfaceRequests.browseExistingList(type, organization_id, token, search)
+			}
+			let selfResourceFilter = {
+				organization_id,
+				status: common.RESOURCE_STATUS_PUBLISHED,
+				published_id: null,
+			}
+			if (type) selfResourceFilter.type = type
+			const selfReources = await resourceQueries.findAll(selfResourceFilter, [
+				'id',
+				'title',
+				'type',
+				'created_by',
+				'created_at',
+			])
+
+			let combinedData =
+				resources?.success && resources?.data?.result?.data?.length > 0
```

Comment:
@adithyadinesh0412  optimize this
let combinedData = [
  ...(resources?.success && resources?.data?.result?.data?.length ? resources.data.result.data : []),
  ...(selfResources.length ? selfResources : [])
];